client/android/WorldScope/app/src/main/java/com/litmus/worldscope/LitmusPlayer.java

```diff
@@ -0,0 +1,284 @@
+        this.audioRenderer = rendererBuilder.getAudioRenderer();
+    }
+
+    // Function to setSurface for video rendering
+    public void setSurface(Surface surface) {
+        this.surface = surface;
+    }
+
+    // Function to getSurface from player
+    public Surface getSurface() {
+        return this.surface;
+    }
+
+    // Function to destroy surface
+    public void blockingClearSurface() {
+        surface = null;
+        pushSurface(true);
+    }
+
+    // Function to push surface to videoRenderer so that videoRenderer can playback
+    private void pushSurface(boolean blockForSurfacePush) {
+        if (videoRenderer == null) {
+            Log.i(TAG, "NO VIDEO RENDERER");
+            return;
+        }
+
+        if(surface == null) {
+            Log.i(TAG, "NO SURFACE");
+            return;
+        }
+
+        Log.i(TAG, "PUSHING");
+
+        if (blockForSurfacePush) {
+            player.blockingSendMessage(
+                    videoRenderer, MediaCodecVideoTrackRenderer.MSG_SET_SURFACE, surface);
+        } else {
+            player.sendMessage(
+                    videoRenderer, MediaCodecVideoTrackRenderer.MSG_SET_SURFACE, surface);
+        }
+    }
+
+    // Function to add a listener that implements LitmusPlayer
+    public void addPlayerEventsListener(PlayerEventsListener listener) {
+        listeners.add(listener);
+    }
+
+    public void setPlayWhenReady(boolean playWhenReady) {
+        player.setPlayWhenReady(playWhenReady);
+    }
+
+    public void release() {
+        player.release();
+    }
+
+    // Functions to implement Exoplayer.Listener
+    public void onPlayerError(ExoPlaybackException error) {
+
+    }
+
+    public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+    }
+
+    public void onPlayWhenReadyCommitted() {
+
+    }
+
+    // Function to get track from ExoPlayer
+    public int getSelectedTrack(int type) {
+        return player.getSelectedTrack(type);
+    }
+
+    // Function to be called when renderers are ready in DashRendererBuilder
+    public void onRenderers(TrackRenderer[] renderers) {
+        for (int i = 0; i < RENDERER_COUNT; i++) {
+            if (renderers[i] == null) {
+                // Convert a null renderer to a dummy renderer.
+                renderers[i] = new DummyTrackRenderer();
+            }
+        }
+        // Complete preparation
+        this.videoRenderer = renderers[TYPE_VIDEO];
+        this.audioRenderer = (MediaCodecAudioTrackRenderer) renderers[TYPE_AUDIO];
+        this.codecCounters = videoRenderer instanceof MediaCodecTrackRenderer
+                ? ((MediaCodecTrackRenderer) videoRenderer).codecCounters
+                : renderers[TYPE_AUDIO] instanceof MediaCodecTrackRenderer
+                ? ((MediaCodecTrackRenderer) renderers[TYPE_AUDIO]).codecCounters : null;
+        Log.i(TAG, "Preparing player");
+        this.player.prepare(this.videoRenderer, this.audioRenderer);
+    }
```

Comment:
Use If-else statements would improve visibility. (If this section was not auto-gen)